permissions/resource_permissions_test.go

```diff
@@ -121,6 +121,167 @@ func TestResourcePermissionsRead_RemovedCluster(t *testing.T) {
 	}.ApplyNoError(t)
 }
 
+func TestResourcePermissionsRead_Mlflow_Model(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		// Pass list of API request mocks
+		Fixtures: []qa.HTTPFixture{
+			me,
+			{
+				Method:   http.MethodGet,
+				Resource: "/api/2.0/permissions/registered-models/fakeuuid123",
+				Response: ObjectACL{
+					ObjectID:   "/registered-models/fakeuuid123",
+					ObjectType: "registered-model",
+					AccessControlList: []AccessControl{
+						{
+							UserName:        TestingUser,
+							PermissionLevel: "CAN_READ",
+						},
+						{
+							UserName:        TestingAdminUser,
+							PermissionLevel: "CAN_MANAGE",
+						},
+					},
+				},
+			},
+		},
+		Resource: ResourcePermissions(),
+		Read:     true,
+		New:      true,
+		ID:       "/registered-models/fakeuuid123",
+	}.Apply(t)
+	assert.NoError(t, err, err)
```

Comment:
Please rewrite with ApplyAndEqual method of resource fixture. It provides more readable tests

Reply:
Thanks for the pointer - I can't find an `ApplyAndEqual` method, do you mean `ApplyAndExpectData`? I'm trying that out but hitting issues with [an assumption in that method](https://github.com/databricks/terraform-provider-databricks/blob/c72250b58ca18883fd62745902cbdd33eb881527/qa/testing.go#L273) that the key of the provided data object is a string array. Thus when I try to pass a `data` argument (representing expected resource data) like

```
	expectedData := map[string]any{
		"registered_model_id": "fakeuuid123",
		"access_control": []any{
			map[string]any{
				"user_name":        TestingUser,
				"permission_level": "CAN_READ",
			},
		},
	}
```

I get a runtime type conversion error:

```
--- FAIL: TestResourcePermissionsRead_Mlflow_Model (0.27s)
panic: interface conversion: interface {} is []interface {}, not []string [recovered]
	panic: interface conversion: interface {} is []interface {}, not []string

goroutine 8 [running]:
testing.tRunner.func1.2({0x18d1900, 0xc00014c390})
	/usr/local/Cellar/go/1.18.4/libexec/src/testing/testing.go:1389 +0x24e
testing.tRunner.func1()
	/usr/local/Cellar/go/1.18.4/libexec/src/testing/testing.go:1392 +0x39f
panic({0x18d1900, 0xc00014c390})
	/usr/local/Cellar/go/1.18.4/libexec/src/runtime/panic.go:838 +0x207
github.com/databricks/terraform-provider-databricks/qa.ResourceFixture.ApplyAndExpectData({{0xc000538aa0, 0x2, 0x2}, 0xc00071a2a0, 0x0, 0x0, 0x0, {0x0, 0x0}, 0x0, ...}, ...)
	/Users/sid.murching/go/src/terraform-provider-databricks/qa/testing.go:274 +0x53d
github.com/databricks/terraform-provider-databricks/permissions.TestResourcePermissionsRead_Mlflow_Model(0x0?)
	/Users/sid.murching/go/src/terraform-provider-databricks/permissions/resource_permissions_test.go:161 +0x528
testing.tRunner(0xc000702b60, 0x1a21d88)
	/usr/local/Cellar/go/1.18.4/libexec/src/testing/testing.go:1439 +0x102
created by testing.(*T).Run
	/usr/local/Cellar/go/1.18.4/libexec/src/testing/testing.go:1486 +0x35f
```

I naively tried the solution in [this StackOverflow post](https://stackoverflow.com/a/18211675/12111347) for asserting deep equality, but found that `access_control` blocks seem to have a different type from what's expected & also contain more fields than I care to assert on:

```
2022/08/01 21:52:11 [DEBUG] GET /api/2.0/preview/scim/v2/Me
2022/08/01 21:52:11 [DEBUG] 200 OK  {
  "userName": "admin"
} <- GET /api/2.0/preview/scim/v2/Me
    /Users/sid.murching/go/src/terraform-provider-databricks/permissions/testing.go:275:
        	Error Trace:	/Users/sid.murching/go/src/terraform-provider-databricks/permissions/testing.go:275
        	            				/Users/sid.murching/go/src/terraform-provider-databricks/permissions/resource_permissions_test.go:161
        	Error:      	Should be true
        	Test:       	TestResourcePermissionsRead_Mlflow_Model
        	Messages:   	&{%!s(schema.SchemaSetFunc=0x1722300) map[1040111181:map[group_name: permission_level:CAN_READ service_principal_name: user_name:ben]] {%!s(uint32=1) {%!s(int32=0) %!s(uint32=0)}}} did not have expected value [map[permission_level:CAN_READ user_name:ben]]
--- FAIL: TestResourcePermissionsRead_Mlflow_Model (0.27s)
FAIL
FAIL	github.com/databricks/terraform-provider-databricks/permissions	1.788s
```

Any suggestions here? I'd prefer not to make major changes to the lower level testing library in this PR as a Golang noob :) but happy to send a follow-up PR to use `ApplyAndExpectData` if it's enhanced to support my use case

Reply:
For good measure I also did some searching and found that `ApplyAndExpectData` is always called on a map from `string -> string[]` or map from `string -> string`, but that seems insufficient for testing the `access_control` blocks in my case

Reply:
@smurching i've realised that i wrote `ApplyAndExpectData` to work properly with sets of strings only. i'm relaxing the requirement on `ApplyAndExpectData` for this pr :)